projects/bp-gallery/src/components/views/geoguessr/GeoView.tsx

```diff
@@ -0,0 +1,154 @@
+import { Box, Button, Modal, Typography } from '@mui/material';
+import { Info } from '@mui/icons-material';
+import { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  useGetAllPicturesByArchiveQuery,
+  useGetPictureGeoInfoQuery,
+  useGetPictureInfoQuery,
+} from '../../../graphql/APIConnector';
+import { useSimplifiedQueryResponseData } from '../../../graphql/queryUtils';
+import { asApiPath } from '../../../helpers/app-helpers';
+import { FlatArchiveTag, FlatPicture } from '../../../types/additionalFlatTypes';
+import ZoomWrapper from '../picture/overlay/ZoomWrapper';
+import GeoMap from './GeoMap';
+import './GeoView.scss';
+
+const getAllPictureIds = (archives: FlatArchiveTag[]) => {
+  const list: string[] = [];
+  archives.forEach(archive => archive.pictures?.forEach(picture => list.push(picture.id)));
+  return list;
+};
+
+const shufflePictureIds = (pictureIds: string[], seed: number) => {
+  const newPictureIds = [...pictureIds];
+  for (let i = newPictureIds.length - 1; i > 0; i--) {
+    const j = Math.abs(Math.floor(i * Math.sin(i % seed)));
+    [newPictureIds[i], newPictureIds[j]] = [newPictureIds[j], newPictureIds[i]];
+  }
+  return newPictureIds;
+};
```

Comment:
maybe instead use the shuffle function from lodash? performance should be similar though